src/callbacks.jl

```diff
@@ -31,6 +31,32 @@ function callback_value(cb_data, x::VariableRef)
     )
 end
 
+"""
+    callback_value(cb_data, expr::GenericAffExpr)
+
+Return the primal solution of an affine expression inside a callback by getting
+the value for each variable appearing in the expression.
+
+`cb_data` is the argument to the callback function, and the type is dependent on
+the solver.
+"""
+function callback_value(cb_data, expr::GenericAffExpr)
+    return expr.constant + sum(callback_value(cb_data, var) * coeff for (var, coeff) in expr.terms)
```

Comment:
You can use `value(expr, v - > callbak_value(cb_data, v))`

Reply:
done thanks. For a future version of JuMP, value should take the function as first argument, it's more idiomatic and allows for the do block syntax

Reply:
Good point, we can open an issue with the breaking tag to remind us of this.